velox/functions/prestosql/Arithmetic.h

```diff
@@ -85,6 +121,33 @@ struct DivideFunction {
   }
 };
 
+template <typename T>
+struct IntervalDivideFunction {
+  FOLLY_ALWAYS_INLINE void
+  call(int64_t& result, const int64_t a, const double b)
+// depend on compiler have correct behaviour for divide by zero
+#if defined(__has_feature)
+#if __has_feature(__address_sanitizer__)
+      __attribute__((__no_sanitize__("float-divide-by-zero")))
+      __attribute__((__no_sanitize__("float-cast-overflow")))
+#endif
+#endif
+  {
+    if UNLIKELY (a == 0 || std::isnan(b) || !std::isfinite(b)) {
+      result = 0;
+      return;
+    }
+    double resultDouble = a / b;
+    if LIKELY (
+        std::isfinite(resultDouble) && resultDouble >= kLongMin &&
```

Comment:
curious can result be finite but < kLongMin or > kLongMax ? (I dont think that should be possible and these additional checks are probably redundant)

Reply:
This is possible. Imagine that a is 9223372036854775807 while b is 0.1. The resultDouble is 92233720368547758070 that is larger than kLongMax (9223372036854775807).